### PR TITLE
refactor the code in func backOffTime()

### PR DESCRIPTION
**Pull request by brilliant-lx**
### Description
no behavior change, only make the code more readable

### CODE REVIEW

File: consensus/parlia/parlia.go [hunk 1]
```diff
@@ -1321,59 +1321,67 @@ func (p *Parlia) backOffTime(snap *Snapshot, header *types.Header, val common.Ad
 	if snap.inturn(val) {
 		return 0
 	} else {
-		idx := snap.indexOfVal(val)
+		delay := initialBackOffTime
+		validators := snap.validators()
+		if p.chainConfig.IsBohr(header.Number) {
+			// reverse the key/value of snap.Recents to get recentsMap
+			recentsMap := make(map[common.Address]uint64, len(snap.Recents))
```
Comment (BetZzzzz) [inline consensus/parlia/parlia.go:1328]: block height is no need here, value filed could just be struct{}

File: consensus/parlia/parlia.go [hunk 2]
```diff
@@ -1321,59 +1321,67 @@ func (p *Parlia) backOffTime(snap *Snapshot, header *types.Header, val common.Ad
...
+					continue
+				}
+				temp = append(temp, addr)
+			}
+			validators = temp
+		}
+
+		// get the index of current validator and its shuffled backoff time.
+		idx := -1
+		for index, itemAddr := range validators {
```
Comment (BetZzzzz) [inline consensus/parlia/parlia.go:1361]: could get rid of this for loop, just calculate the `idx` in the loop where calculating updated validators(line:1350)

File: consensus/parlia/parlia.go (hunk 1 above)
Comment (brilliant-lx) [reply consensus/parlia/parlia.go:1328]: actually, it is used bellow for debug purpose.

File: consensus/parlia/parlia.go (hunk 2 above)
Comment (brilliant-lx) [reply consensus/parlia/parlia.go:1361]: but Bohr could have not reached.

**Review by BetZzzzz — APPROVED** [review]

**Comment by NathanBSC** [comment]
lgtm

**Review by buddh0 — APPROVED** [review]

**Pull request merged**